File: src/main/java/legacy/kms/chapter01/Theater.java
package legacy.kms.chapter01;

public class Theater {

    private final TicketSeller ticketSeller;

    public Theater(TicketSeller ticketSeller) {
        this.ticketSeller = ticketSeller;
    }

    public void enter(Audience audience) {
        if (audience.hasInvitation()) {
            Ticket ticket = ticketSeller.getTicket();
            audience.setTicket(ticket);
        } else {
            Ticket ticket = ticketSeller.getTicket();
            audience.minusAmount(ticket.getFee());
            ticketSeller.plusAmount(ticket.getFee());
            audience.setTicket(ticket);
        }
    }
}

File: src/main/kotlin/hoondong/Theater.kt
package hoondong

class Theater(private val ticketSeller: TicketSeller) {
    fun enter(audience: Audience) {
        if (audience.getBag().hasInvitation()) {
            val ticket: Ticket = ticketSeller.getTicketOffice().getTicket()
            audience.getBag().setTicket(ticket)
        } else {
            val ticket = ticketSeller.getTicketOffice().getTicket()
            audience.getBag().minusAmount(ticket.fee)
            ticketSeller.getTicketOffice().plusAmount(ticket.fee)
            audience.getBag().setTicket(ticket)
        }
    }
}